src/python/pants/backend/python/rules/interpreter_constraints.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Awesome! I recommend breaking this out into a dedicated PR as prework. Will make this PR easier to review.

Reply:
https://github.com/pantsbuild/pants/pull/8410 is just the interpreter constraints stuff. I'll shortly modify this PR to be based on the git commit there.

Reply:
although, hm, it seems like stu thinks this isn't worth it to do as a separate rule, and I think I agree. I'm gonna close this and try to re-implement this functionality as methods on `ConsoleRule`.

Reply:
I agree with him to not have it be a separate rule. Thanks!